addons/google_calendar/models/calendar.py

```diff
@@ -109,22 +109,29 @@ def _odoo_attendee_commands(self, google_event):
         if google_event.exists(self.env):
             existing_attendees = self.browse(google_event.odoo_id(self.env)).attendee_ids
         attendees_by_emails = {tools.email_normalize(a.email): a for a in existing_attendees}
-        for attendee in google_attendees:
-            email = attendee.get('email')
-
+        partners = self.env['mail.thread']._mail_find_partner_from_emails(emails, records=self, force_create=True, extra_domain=[('type', '!=', 'private')])
+        for attendee in zip(emails, partners, google_attendees):
+            email = attendee[0]
+            state = attendee[2].get('status', {}).get('response', 'accepted')
             if email in attendees_by_emails:
                 # Update existing attendees
-                attendee_commands += [(1, attendees_by_emails[email].id, {'state': attendee.get('responseStatus')})]
+                attendee_commands += [(1, attendees_by_emails[email].id, {'state': state})]
             else:
                 # Create new attendees
-                partner = self.env.user.partner_id if attendee.get('self') else self.env['res.partner'].find_or_create(attendee.get('email'))
-                attendee_commands += [(0, 0, {'state': attendee.get('responseStatus'), 'partner_id': partner.id})]
+                if attendee[2].get('self'):
+                    partner = self.env.user.partner_id
+                elif attendee[1]:
```

Comment:
@thomaspaulb basically this elif resolves the falsy partner problem. Just need to test it. I will do the same for microsoft_calendar, almost done, will create an additional PR for that